Sources/Apodini/Configurations/DependentStaticConfigurationBuilder.swift

```diff
@@ -0,0 +1,48 @@
+//                   
+// This source file is part of the Apodini open source project
+//
+// SPDX-FileCopyrightText: 2019-2021 Paul Schmiedmayer and the Apodini project authors (see CONTRIBUTORS.md) <paul.schmiedmayer@tum.de>
+//
+// SPDX-License-Identifier: MIT
+//              
+
+/// A function builder used to aggregate multiple `DependentStaticConfiguration`s
+@resultBuilder
+public enum DependentStaticConfigurationBuilder<ParentConfiguration: ConfigurationWithDependents> {
+    
+    /// A method that transforms multiple `DependentStaticConfiguration`s
+    ///
+    /// - Parameter staticConfigurations: A variadic number of `DependentStaticConfiguration`
+    ///
+    /// - Returns: An `AnyConfigurationCollection` which consists of `ConfigurationConvertible`s
+    public static func buildBlock(_ staticConfigurations: [AnyDependentStaticConfiguration]...) -> [AnyDependentStaticConfiguration] {
```

Comment:
I think using variadic parameters and in generally `AnyDependentStaticConfiguration` works in a way that one can pass `ConfigurationWithDependents` to the builder but not only restricting them to a specific configuration type by constraining it to a specific `InternalConfiguration`.
I think that `ParentConfiguration` does not need to conform to `ConfigurationWithDependents` and that the constraint should then be used with concrete `ConfigurationWithDependents` in the build functions using the Swift `where` clause. Otherwise one can pass any `ConfigurationWithDependents` to the builder from the outside and we have the same level of type safety as we have internally using the type erased version.

```swift
@resultBuilder
public struct DependentStaticConfigurationBuilder<Configuration> {
    public static func buildIf<T: ConfigurationWithDependents>(_ component: T?) -> T? where T.Configuration == Configuration {
        return component
    }
    
    public static func buildBlock<T: ConfigurationWithDependents>(_ arg0: T) -> T where T.Configuration == Configuration {
        return arg0
    }

    public static func buildEither<T: ConfigurationWithDependents>(first: T) -> T where T.Configuration == Configuration {
        return first
    }
    
    public static func buildEither<T: ConfigurationWithDependents>(second: T) -> T where T.Configuration == Configuration {
        return second
    }

    public static func buildIf<T: ConfigurationWithDependents>(first: T) -> T where T.Configuration == Configuration {
        return first
    }

    public static func buildBlock<T0: ConfigurationWithDependents, T1: ConfigurationWithDependents >(_ arg0: T0, _ arg1: T1) -> [AnyDependentStaticConfiguration] where T0.Configuration == Configuration, T1.Configuration == Configuration {
        return [arg0, arg1]
    }
}
```
We will need more buildBlock functions with more types, as [Swift sadly does not yet support variadic generics (but this is on the way)](https://forums.swift.org/t/variadic-generics/54511)

Reply:
resultBuilders allow for the conversion to an intermediate type using the `buildExpression` method, as explained [here](https://docs.swift.org/swift-book/ReferenceManual/Attributes.html#ID634). All subsequent methods can take `[AnyDependentStaticConfiguration]`s, as the correct type was already ensured by `buildExpression`. This can be verified by trying to pass an `OpenAPI` Configuration into a `HTTP` Configuration, which will not compile. Swift is actually being quite handy with types here for once!

Reply:
Nice, learned something new, very cool! Thanks for sharing the link and the insight 🚀